include/PriorityQueue.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+#include "Dog.h"
+
+struct MyComparator {
+    template<typename T>
+    bool operator()(const T& a, const T& b) const {
+        return  (b - a) < 0;
+    }
+};
+
+template<typename T, typename Comparator = MyComparator>
+class PriorityQueue {
+public:
+    void push(const T& t) {
+        typename std::list<T>::iterator it = elements.begin();

```

Comment:
When possible it's recommended to call cbegin() and cend(). You can never be too safe